components/forms/src/api.rs

```diff
@@ -0,0 +1,248 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::error::*;
+use rusqlite::{Connection, NO_PARAMS, Transaction};
+use serde::Serialize;
+use serde_derive::*;
+use sql_support::{self, ConnExt};
+use std::time;
+use sync_guid::Guid;
+
+#[derive(Debug, Clone, Hash, PartialEq, Serialize, Deserialize, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct Address {
+    #[serde(rename = "id")]
+    pub guid: Guid,
+
+    pub given_name: String,
+
+    pub additional_name: String,
+
+    pub family_name: String,
+
+    #[serde(default)]
+    pub organization: String,
+
+    #[serde(default)]
+    pub street_address: String,
+
+    pub address_level3: String,
+
+    pub address_level2: String,
+
+    pub address_level1: String,
+
+    pub postal_code: String,
+
+    pub country: String,
+
+    #[serde(default)]
+    pub tel: String,
+
+    #[serde(default)]
+    pub email: String,
+
+    #[serde(default)]
+    #[serde(deserialize_with = "deserialize_timestamp")]
+    pub time_created: i64,
```

Comment:
I think I'd rather see a distinct type for timestamp fields - it's worked out quite well in places, and I wish I'd done that in webext-storage, particularly when integrating with desktop.

Reply:
Are you referring to [this](https://github.com/mozilla/application-services/blob/673e700194cfa44ad3c8629f74a2b01844cea7c3/components/places/src/types.rs)?

Reply:
Yep!

Reply:
Since this would be used in more than one component should it the types logic be moved out of the places component? Would [support](https://github.com/mozilla/application-services/tree/673e700194cfa44ad3c8629f74a2b01844cea7c3/components/support) be a good place for it?

Reply:
`support` sounds perfect!